threeML/plugins/ogip.py

```diff
@@ -748,8 +750,24 @@ def _read_matrix(self, data, n_channels):
             m_start = 0
 
             for j in range(n_grp[i]):
+                # print '-----------'
+                # print f_chan[i][j]
+                # print n_chan[i][j]
+                #
+                # print len(rsp[i, f_chan[i][j]: f_chan[i][j] + n_chan[i][j]])
+                # print len(matrix[i][m_start:m_start + n_chan[i][j]])
+                # print '-----------'
+                #
+                try:
+
+                    rsp[i, f_chan[i][j]: f_chan[i][j] + n_chan[i][j]] = matrix[i][m_start:m_start + n_chan[i][j]]
+
+                except(ValueError):
+
+                    rsp[i, f_chan[i][j] -1: f_chan[i][j]-1 + n_chan[i][j]] = matrix[i][m_start:m_start + n_chan[i][j]]
+
+
```

Comment:
I don't like this modification, as the response is read in a non-compliant way.

Do you have a matrix which was not read by the previous implementation. while it should have? If yes, please open an issue and provide the matrix, instead of modifying it here. 

This is a very critical part, we need to be extremely careful, so please revert to the original implementation and let's talk about this in the issue you are going to open.